fraudserv/service.go

```diff
@@ -177,7 +177,15 @@ func (f *ProofService[H]) processIncoming(
 	proofType fraud.ProofType,
 	from peer.ID,
 	msg *pubsub.Message,
-) pubsub.ValidationResult {
+) (res pubsub.ValidationResult) {
+	defer func() {
+		err := recover()
+		if err != nil {
+			log.Errorf("PANIC while processing a proof: %s", err)
```

Comment:
Pls Errorw. err might not have Stringer implementaiton

Reply:
nvm, %s apparently looks for Error() method on top of String() too.

Reply:
The err returned from recover is actually a string and not error interface